Emojigrams/Emojigrams/EmojigramGuessProgramaticLayoutViewController.swift

```diff
@@ -51,6 +51,7 @@ class EmojigramGuessProgramaticLayoutViewController: EmojigramGuesserBaseViewCon
         textField.font = UIFont.systemFont(ofSize: 14, weight: .regular)
         textField.translatesAutoresizingMaskIntoConstraints = false
         textField.borderStyle = .line
+        textField.delegate = self
```

Comment:
#? does this class extend from the base class? if not, can we access textfield delegate here, maybe we should have this line in the base class which inherits from textfielddelegate?

Reply:
Yep, once you extend a parent class, all children also share in that extension. The code knows during compilation that the child class now also has the extensions as well and so EmojigramGuessProgramaticLayoutVC is considered to be a UITextFieldDelegate as well.